kanin/src/app/task.rs

```diff
@@ -126,13 +130,9 @@ async fn handle_request<H, S, Args, Res>(
             let mut props = BasicProperties::default();
 
             if let Some(correlation_id) = correlation_id {
-                props = props.with_correlation_id(correlation_id);
+                props = props.with_correlation_id(correlation_id.clone());
```

Comment:
Why the clone? I don't see it being used elsewhere

Reply:
`with_correlation_id` takes ownership but the id is only a reference, that's why. There was also a clone previously if you check the diff above.